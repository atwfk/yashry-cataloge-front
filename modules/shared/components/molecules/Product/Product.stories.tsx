import type { ReactElement } from "react";
import type { Story } from "@storybook/react";
import Product from "./Product";
import { IData } from "@modules/shared/types/IData";

const args = {
  title: "Molecules/Product",
  component: Product,
};

export default args;

const Template: Story<IData.IProduct> = (args): ReactElement => (
  <div className="flex">
    <Product {...args} />
  </div>
);

export const product = Template.bind({});
product.args = {
  price: 614,
  id: 1,
  name: "andcrafted Cotton Keyboard",
  image: "http://placeimg.com/640/480/cats",
  rating: 4,
};
