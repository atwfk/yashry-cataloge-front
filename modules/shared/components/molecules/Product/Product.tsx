import React from "react";
import type { FC, ReactElement } from "react";
import type { IData } from "../../../../shared/types/IData";
import Image from "next/image";
import Box from "../../atoms/Box/Box";
import P from "../../atoms/Paragraph/Paragraph";
import Stars from "../../atoms/Stars/Stars";

const Product: FC<IData.IProduct> = ({
  price,
  rating,
  id,
  name,
  image,
  currency,
}): ReactElement => {
  return (
    <Box>
      <>
        <div className="w-full">
          <Image
            src={image}
            alt="product"
            layout="responsive"
            objectFit="cover"
            width={300}
            height={300}
            id={`${id}`}
          />
        </div>
        <P fontColor="dark" fontSize="md" fontWeight="medium">
          {name}
        </P>
        <Stars rating={rating} />
        <P fontColor="accent" fontSize="sm" fontWeight="normal">
          <span className="mr-1">{currency}</span>
          <span>${price}</span>
        </P>
      </>
    </Box>
  );
};

export default Product;
