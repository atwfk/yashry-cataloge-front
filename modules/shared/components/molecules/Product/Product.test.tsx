import React from "react";
import Product from "./Product";
import renderer from "react-test-renderer";

describe("Product", () => {
  it("should render single product", () => {
    const product = {
      id: 1,
      name: "Handcrafted Cotton Keyboard",
      image: "http://placeimg.com/640/480/cats",
      color: "black",
      price: 614,
      currency: "USD",
      releaseDate: "2021-09-23T08:15:36.195Z",
      categoryId: 2,
      rating: 4,
    };

    const tree = renderer
      .create(
        <Product
          price={product.price}
          rating={product.rating}
          id={product.id}
          name={product.name}
          image={product.image}
          color={product.color}
          currency={product.currency}
          releaseDate={product.releaseDate}
          categoryId={product.categoryId}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
