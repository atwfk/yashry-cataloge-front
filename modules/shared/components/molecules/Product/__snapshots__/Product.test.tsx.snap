// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should render single product 1`] = `
<div
  className="box"
>
  <div
    className="w-full"
  >
    <div
      style={
        Object {
          "boxSizing": "border-box",
          "display": "block",
          "margin": 0,
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "paddingTop": "100%",
          }
        }
      />
      <img
        alt="product"
        data-nimg="responsive"
        decoding="async"
        id="1"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style={
          Object {
            "border": "none",
            "bottom": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": 0,
            "left": 0,
            "margin": "auto",
            "maxHeight": "100%",
            "maxWidth": "100%",
            "minHeight": "100%",
            "minWidth": "100%",
            "objectFit": "cover",
            "objectPosition": undefined,
            "padding": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 0,
          }
        }
      />
      <noscript>
        <img
          alt="product"
          data-nimg="responsive"
          decoding="async"
          id="1"
          loading="lazy"
          sizes="100vw"
          src="/_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=3840&q=75"
          srcSet="/_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=640&q=75 640w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=750&q=75 750w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=828&q=75 828w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=1080&q=75 1080w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=1200&q=75 1200w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=1920&q=75 1920w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=2048&q=75 2048w, /_next/image?url=http%3A%2F%2Fplaceimg.com%2F640%2F480%2Fcats&w=3840&q=75 3840w"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
      </noscript>
    </div>
  </div>
  <p
    className="text-base text-dark font-medium text-center"
  >
    Handcrafted Cotton Keyboard
  </p>
  <div
    className="flex"
  >
    <div
      className="w-8"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="star"
          data-nimg="responsive"
          data-testid="filled-stars"
          decoding="async"
          id="0"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="star"
            data-nimg="responsive"
            data-testid="filled-stars"
            decoding="async"
            id="0"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </div>
    <div
      className="w-8"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="star"
          data-nimg="responsive"
          data-testid="filled-stars"
          decoding="async"
          id="1"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="star"
            data-nimg="responsive"
            data-testid="filled-stars"
            decoding="async"
            id="1"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </div>
    <div
      className="w-8"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="star"
          data-nimg="responsive"
          data-testid="filled-stars"
          decoding="async"
          id="2"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="star"
            data-nimg="responsive"
            data-testid="filled-stars"
            decoding="async"
            id="2"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </div>
    <div
      className="w-8"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="star"
          data-nimg="responsive"
          data-testid="filled-stars"
          decoding="async"
          id="3"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="star"
            data-nimg="responsive"
            data-testid="filled-stars"
            decoding="async"
            id="3"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Femoji%2F36%2F000000%2Fstar-emoji.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </div>
    <div
      className="w-8"
    >
      <div
        style={
          Object {
            "boxSizing": "border-box",
            "display": "block",
            "margin": 0,
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "boxSizing": "border-box",
              "display": "block",
              "paddingTop": "100%",
            }
          }
        />
        <img
          alt="star"
          data-nimg="responsive"
          data-testid="not-filled-stars"
          decoding="async"
          id="4"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": undefined,
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="star"
            data-nimg="responsive"
            data-testid="not-filled-stars"
            decoding="async"
            id="4"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=3840&q=75"
            srcSet="/_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fimg.icons8.com%2Fcolor%2F36%2F000000%2Fstar--v1.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </div>
    </div>
  </div>
  <p
    className="text-sm text-accent font-normal"
  >
    <span
      className="mr-1"
    >
      USD
    </span>
    <span>
      $
      614
    </span>
  </p>
</div>
`;
